### ISSUE TITLE
ScaleInvariantSignalNoiseRatio from spectrogram 

### ISSUE THREAD

**Issue by walidbou6** [PyTorchLightning/metrics#897]
## 🚀 Feature

Scale Invariant Signal Noise Ratio logic computed with waves data, but in some model training cases we work on predicting the spectrogram from a given data so to get the performance of your training you should invert the estimated spectrogram to wav and run the SI_SNR on it, it would be great if we can compute the SI_SNR direct with the spectrogram  

check [here](https://nussl.github.io/docs/_modules/nussl/ml/train/loss.html#SISDRLoss) I believe they have this option

**Comment by SkafteNicki** [comment]
cc: @quancs any opinion on this?

**Comment by quancs** [comment]
@walidbou6 Hello, I checked you link. Seems the code is not clear how it is relevant to spectrogram, as the input shape is [num_batch, num_samples, num_sources] not containing the frequency dimension. It would be great if you can elaborate the SI_SNR on spectrogram ^_^

**Comment by walidbou6** [comment]
let me tell you a funny story, I wanted to train a model to predict a mask for speech enhancement, done building the model and the training logic and for the loss metric I wanted to use SI_SNR or SI_SDR, but those are using wave data to compute the model performance, so I needed to invert spectrogram to wav to preform the calcultion and here where the problems started, I was using librosa to generat sepctrograms (STFT, ISTFT) so to invert to wav I needed to detach it once I do that the model throw an error "no grad for loss" so I have added the requierd grad to loss, worked for few steps and throw OOM error, here where I just give the `nussl.ml.train.loss.SISDRLoss` the predicted spectrogram with the target spectrogram and worked fine, the funny part the model give me  a really good results, I have checked now and found that logic it's not right.

**Comment by quancs** [comment]
Seems you find a new land... Maybe you should carefully check if it could be your paper  ^_^

For your case, you can try the STFT and iSTFT provided by torch, which would not drop the grad.
